monkey/infection_monkey/network/info.py

```diff
@@ -21,6 +22,41 @@
 RTF_REJECT = 0x0200
 
 
+class IntRange:
+    """
+    Represents an inclusive range of integers, with a step size of 1.
+
+    Ensures that min <= max.
+    """
+
+    def __init__(self, min: int, max: int):
+        self._min = min
+        self._max = max
+        if min > max:
+            self._min, self._max = max, min
+
+    @property
+    def max(self) -> int:
+        """The maximum value in the range."""
+        return self._max
+
+    @property
+    def min(self) -> int:
+        """The minimum value in the range."""
+        return self._min
+
+
+def port_range(range: IntRange) -> Iterator[NetworkPort]:
+    """Yields port values in the provided range, bounded by [0, 65535]."""
+    min_ = max(0, range.min)
+    max_ = min(65535, range.max) + 1
+    current = min_
+
+    while current < max_:
+        yield NetworkPort(current)
+        current += 1
```

Comment:
```suggestion
    return range(min_, max_)
```

Reply:
The purpose of this generator is to yield `NetworkPort`s. If it returns a `range`, mypy complains (unless I don't document the return value)

Reply:
Oh, because `range()` returns an `Iterator[int]` or similar. My bad. Try 
```suggestion
    return map(NetworkPort, range(min_, max))
```

Reply:
That will work!